hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotDiffManager.java

```diff
@@ -282,35 +278,17 @@ public PersistentMap<String, SnapshotDiffJob> getSnapDiffJobTable() {
     return snapDiffJobTable;
   }
 
-  private Optional<ManagedSSTDumpTool> initSSTDumpTool(
+  private boolean initSSTDumpTool(
       final OzoneConfiguration conf) {
     if (conf.getBoolean(OMConfigKeys.OZONE_OM_SNAPSHOT_LOAD_NATIVE_LIB,
         OMConfigKeys.OZONE_OM_SNAPSHOT_LOAD_NATIVE_LIB_DEFAULT)) {
       try {
-        int threadPoolSize = conf.getInt(
-                OMConfigKeys.OZONE_OM_SNAPSHOT_SST_DUMPTOOL_EXECUTOR_POOL_SIZE,
-                OMConfigKeys
-                    .OZONE_OM_SNAPSHOT_SST_DUMPTOOL_EXECUTOR_POOL_SIZE_DEFAULT);
-        int bufferSize = (int) conf.getStorageSize(
-            OMConfigKeys.OZONE_OM_SNAPSHOT_SST_DUMPTOOL_EXECUTOR_BUFFER_SIZE,
-            OMConfigKeys
-                .OZONE_OM_SNAPSHOT_SST_DUMPTOOL_EXECUTOR_BUFFER_SIZE_DEFAULT,
-                StorageUnit.BYTES);
-        this.sstDumpToolExecService = Optional.of(new ThreadPoolExecutor(0,
-                threadPoolSize, 60, TimeUnit.SECONDS,
-                new SynchronousQueue<>(), new ThreadFactoryBuilder()
-            .setNameFormat(ozoneManager.getThreadNamePrefix() +
-                "snapshot-diff-manager-sst-dump-tool-TID-%d")
-                .build(),
-                new ThreadPoolExecutor.DiscardPolicy()));
-        return Optional.of(new ManagedSSTDumpTool(sstDumpToolExecService.get(),
-            bufferSize));
+        return ManagedRawSSTFileReader.loadLibrary();
       } catch (NativeLibraryNotLoadedException e) {
-        this.sstDumpToolExecService.ifPresent(exec ->
-            closeExecutorService(exec, "SstDumpToolExecutor"));
+        return false;
```

Comment:
It would be good to log the exception as warn or something in case loading fails. Otherwise one would never know that optimize diff is not working even when native lib is enabled.

Reply:
done